### PR TITLE
Code Refactoring + Edits

### PR DESCRIPTION
**Pull request by CodeByAidan**
Code Refactoring + Edits involving a new requirement change.

### CODE REVIEW

**Comment by timof121** [comment]
> SIT THE HELL DOWN BOY LETS GO IM BETTER AT CODING

im going to rape you

**Comment by terrified9** [comment]
https://api-minecraft.net/xatsn5m9.png why?

**Comment by livxy** [comment]
> > SIT THE HELL DOWN BOY LETS GO IM BETTER AT CODING
> 
> im going to rape you

mmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmgdkfjhjkhfgddddddddddddhjklasdhgjkadsaaaaaaaaiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiioijkfvbjknhvfkdjhhbgkvbffbbbbv

sorry my dog ate my keyboard

**Comment by livxy** [comment]
NO CONFLICTS? SEE QOFT?!!?!? NO CONFLICTS 

:trollface:

**Comment by terrified9** [comment]
cool

**Comment by CodeByAidan** [comment]
> https://api-minecraft.net/xatsn5m9.png why?

Here @timof121! Let me explain why the hell 73982361305c7a1d553cb79f7eb728a754722e9e is extremely important. 

## What was wrong?
**An empty list was used as a default argument to a function**. Let's use an example, something like:
```py
def compute_patterns(inputs=[]):
    inputs.append("some stuff")
    patterns = ["a list based on"] + inputs
    return patterns
```
if you run this function multiple times, you'll get different results:
```py
>>> compute_patterns()
['a list based on', 'some stuff']  # Expected
>>> compute_patterns()
['a list based on', 'some stuff', 'some stuff']  # VERY BAD VERY YOU SEE BAD BAD BAD
```
``inputs=[]`` **was the part that was causing this mess**

## The Problem
So, in Python, when passing a mutable value as a default argument in a function, **the default argument is mutated anytime that value is mutated.**

Here, "mutable value" refers to anything such as a list, a dictionary or even a class instance.

 ## What happened, exactly?
It's not really straight-forward to see why the above fact can be a problem...

Here's what happens in detail using another example:
```py
def append(element, seq=[]):
    seq.append(element)
    return seq
```
```py
>>> append(1)  # seq is assigned to []
[1]  # This returns a reference the *same* list as the default for `seq`
>>> append(2)  # => `seq` is now given [1] as a default!
[1, 2]  #wtf????!?!??!?!
```
As you can see, Python "retains" the default value and ties it to the function in some way. Since it is mutated inside the function, it is also mutated as a default. In the end, **we use a different default every time the function is called**.

## The solution
Long story short, the solution is quite easy (and tedious :trollface: ).

**Use ``None`` as a default and assign the mutable value inside the function.**

So instead, do this:
```py
# 👇
def append(element, seq=None):
    if seq is None:  # 👍
        seq = []
    seq.append(element)
    return seq
```

```py
>>> append(1)  # `seq` is assigned to []
[1]
>>> append(2)  # `seq` is assigned to [] again!
[2]  # Yay!
```

**Comment by terrified9** [comment]
what the fuck

**Comment by terrified9** [comment]
please kill yourself

**Comment by livxy** [comment]
and that is how i met your mother @timof121

**Comment by terrified9** [comment]
image

**Comment by terrified9** [comment]
> image

explains it all

**Comment by livxy** [comment]
> > image
> 
> explains it all

BRO 💀

**Comment by qoft** [comment]
you made this over complicated + you do this to every repo of mine so hope off my dick

**Comment by livxy** [comment]
> you made this over complicated + you do this to every repo of mine so hope off my pee pee

love you too <3

**Comment by bennyscripts** [comment]
> you made this over complicated + you do this to every repo of mine so hope off my dick

qoft dont listen to them <3

**Comment by CodeByAidan** [comment]
This PR was closed a year ago. Please consider opening a [new issue](https://github.com/qoft/sellapp/issues/new/choose) instead of leaving a comment here.

we still have beef @qoft
